packages/react/src/components/DateTimePicker/dateTimePickerUtils.js

```diff
@@ -668,5 +668,103 @@ export const useDateTimePickerTooltip = ({ isExpanded }) => {
     }
   };
 
-  return [isTooltipOpen, toggleTooltip];
+  return [isTooltipOpen, toggleTooltip, setIsTooltipOpen];
 };
+
+/**
+ * Utility to extract time value
+ * @param {Object} value: an object containing at least:
+ *   timeRangeKind: see PICKER_KINDS
+ *   timeSingleValue: time value for single time picker
+ *   timeRangeValue: time value for range time picker
+ * @returns string single or range time value from a given object
+ */
+export const getTimeValue = (value) => {
+  if (value.timeRangeKind === PICKER_KINDS.SINGLE) {
+    return value.timeSingleValue;
+  }
+
+  return value.timeRangeValue;
+};
+
+/**
+ * Hook to validate event and revoke callback
+ * @param {function} closeDropdownCallback: function that will be called if validation passes
+ * @returns void
+ */
+export const useDateTimePickerClickOutside = (closeDropdownCallback) => (evt) => {
+  if (
+    evt?.target.classList?.contains(`${iotPrefix}--date-time-picker__listitem--custom`) ||
+    evt?.target.classList?.contains(`${iotPrefix}--date-time-picker__menu-btn-back`)
+  ) {
+    return;
+  }
+
+  // Composed path is needed in order to detect if event is bubbled from TimePickerSpinner which is a React Portal
+  if (
+    evt.composed &&
+    evt.composedPath().some((el) => el.classList?.contains(`${iotPrefix}--time-picker-spinner`))
+  ) {
+    return;
+  }
+
+  closeDropdownCallback();
+};
+
+/**
+ * Hook to close time picker dropdown and reset default value
+ * @param {Object} object: an object containing:
+ *   isExpanded: current state of the dropdown
+ *   setIsExpanded: useState callback
+ *   setIsCustomRange: useState callback
+ *   defaultValue: props value for time picker
+ *   parseDefaultValue: parses value from string to time picker format
+ *   setCustomRangeKind: useState callback
+ *   dateTimePickerBaseValue: base value for time picker
+ *   setCurrentValue: useState callback
+ *   toLabel: label string
+ *   setHumanValue: parses time picker value to readable format
+ * @returns void
+ */
+export const useCloseDropdown = ({
```

Comment:
Just a question here. Do we reset to default when click outside of the drop down or should we apply the change we have so far?

Reply:
It will discard changes and put default value, regardless of previously saved value. For example, if user saves value on the first edit, but closes on click outside on the second edit, then it will reset to default, not to previously saved value. CC @JordanWSmith15

Reply:
@erzhan-temir-mamyrov What you said is correct. Clicking outside would close the drop down and retain the last saved value (If there was an existing value before the user opened the dropdown, that value would be retained; we should not set it to null).

Reply:
@JordanWSmith15  @erzhan-temir-mamyrov I'm a little confused by the explanation.  When you say "edit" do you mean edits in the same dialog?  And when you say "save" do you mean the "apply" which closes dialog?

> It will discard changes and put default value, regardless of previously saved value.

I don't think that is correct.  ie, a "saved value" should be the new default value when the dialog opens a second time.  (ie, when you "save" then that value has been communicated to and it should now be the new default value.)

Unless I'm misunderstanding something here

Reply:
@sls-ca I fixed the click outside behavior. Now, if dropdown was closed by click outside it will restore previously saved value (any changes saved by clicking Apply button). If saved value doesn't exists then it will restore default value provided from props, if any.